MATRIX_GUIDELINES.md

```diff
@@ -0,0 +1,129 @@
+
+## Core Constraint
+
+**Limited GPU resources are the primary constraint that forces us to keep the CI matrix limited.** GPU runners are a shared, finite resource across all RAPIDS projects. Every matrix entry that requires a GPU runner consumes time on this shared pool. We must strive for maximum coverage within the GPU resource budget.
+
+## Matrix Dimensions
+
+Test matrices span multiple dimensions:
+
+- **CPU Architecture** (`ARCH`): `amd64`, `arm64`
+- **CUDA Version** (`CUDA_VER`): e.g., `12.2.2`, `12.9.1`, `13.0.1`
+- **Python Version** (`PY_VER`): e.g., `3.10`, `3.11`, `3.12`, `3.13`
+- **GPU Architecture** (`GPU`): `l4`, `a100`, `h100`
+- **Driver Version** (`DRIVER`): `earliest`, `latest`
+- **Linux Distribution** (`LINUX_VER`): `rockylinux8`, `ubuntu22.04`, `ubuntu24.04`
+- **Dependencies** (`DEPENDENCIES`): `oldest`, `latest`
+
+## Two-Tier Matrix Strategy
+
+### Pull Request (PR) Matrix
+**Goal**: Fast feedback with focused coverage
+
+- **Minimal size** to provide quick CI results
+- Try to keep the matrix size constant when adding new versions
+  - e.g. if adding a new CUDA or new Python, rearrange existing jobs to maximize coverage while keeping the number of jobs constant
+- Focus on:
+  - Endpoints
+    - Latest versions of everything (newest Python, CUDA, driver, dependencies)
+    - Earliest supported versions of everything (oldest Python, CUDA, driver, dependencies)
+  - "Off-diagonal" elements
+    - We want to test combinations like "oldest CUDA, newest Python" and vice versa
+    - Use different matrices for conda and wheel CI jobs -- we often use wheel jobs to hit some of these "edge" configurations
+  - Broad coverage
+    - Both CPU architectures (`amd64` and `arm64`)
+    - Cover all possible GPU architectures, while respecting the relative pool sizes of the runners
+      - Allocate fewer CI jobs to the pools with fewer runners
+    - Cover oldest and latest drivers, but make sure the driver/CUDA versions are supported by the desired operating system
+
+### Nightly Matrix
+**Goal**: Comprehensive coverage across all supported configurations
+
+- **Expanded matrix** to catch edge cases and combinations not tested in PRs
+- We have a weak preference for nightlies to be a *superset* of the PR matrix, meaning it is the same with additional elements
+- Otherwise, same rules as for the PR matrix: hit the endpoints, off-diagonal elements, and shoot for broad coverage
+
+## Coverage Priorities
+
+When trading off coverage for resource utilization, prioritize in this order:
+
+### 1. CPU Architecture Coverage
+**Goal**: Every PR and nightly build must test both `amd64` and `arm64`
+
+- Both architectures must be represented in PR tests
+- Failures on either architecture are equally important
+
+### 2. CUDA Version Coverage
+**Goal**: Test minimum supported, the latest stable version, and another intermediate version
```

Comment:
We should document here that our build matrix for CUDA versions is only the latest of each major version we support